src/js/material.ts

```diff
@@ -388,6 +405,37 @@ export function MaterialMixin<
 
             return checks;
         }
+
+        /**
+         * @summary Returns a config to create a material from a CIF or POSCAR file.
+         */
+        static getMaterialFileConfig(
+            fileName: string,
+            fileContent: string,
+            fileExtension: string,
+        ): MaterialFileConfig {
+            const config: MaterialFileConfig = {
+                src: {
+                    extension: fileExtension,
+                    filename: fileName,
+                    text: fileContent,
+                    hash: CryptoJS.MD5(fileContent).toString(),
+                },
+            };
+            // add ICSD ID if present
+            if (fileExtension === "cif") {
```

Comment:
Let's simplify and remove the external and icsdId parts handling

Reply:
Done